docs/docs/contribute/environment-setup.mdx

```diff
@@ -0,0 +1,82 @@
+# Setup your Development Environment
+
+Setting up your local copy of the project for development and testing is essential. Follow the steps below to configure your environment.
+
+:::info "Install prerequisites"
+
+Before you start contributing to this project, you will need **Node.js@18** and **npm@8** installed.
+
+:::
+
+## Get the Local Version of the CLI
+
+### Setting up your GitHub Fork
+
+1. Navigate to the [CLI for Microsoft 365](https://github.com/pnp/cli-microsoft365) repository.
+2. Fork this repository. On the GitHub repository page, click the "Fork" button to create your own copy of the repository.
+    - Make sure the option 'Copy the `main` branch only' is selected.
+3. Clone the forked repository to your local machine using Git. Execute the following command in your terminal: `git clone [repository URL]`.
+
+> For more insights about forking a repository, refer to the [GitHub Docs - Fork a repo](https://docs.github.com/en/get-started/quickstart/fork-a-repo#forking-a-repository) guide.
+
+### Setting up your Local Project
+
+After you've cloned your fork onto your device, navigate to the project directory and execute the following commands to get the project running.
+
+1. Run `npm i` to restore all project dependencies.
+2. Run `npm run build` to build the entire project.
+3. Run `npm link` to create a link/reference to your local project. This allows you to reference your locally installed CLI instead of the npm-hosted package.
+
+That's it! If you run `m365 version` now, you will see that you are using the beta version of CLI for Microsoft 365 in your shell!
+
+:::tip
+
+If you have installed the CLI globally using the `npm i -g @pnp/cli-microsoft365` command, we recommend that you uninstall it before running `npm link`.
+
+:::
+
+## Visual Studio Code extensions
+
+Although you can use any IDE, we recommend using Visual Studio Code. When using VS Code, the following extensions will come in handy:
+
+Extension | Why is it useful?
```

Comment:
We should also include `Test Explorer UI` extension in this list. The `Mocha Test Explorer` extension extends the functionality of the `Test Explorer UI` to support `mocha` tests.

https://marketplace.visualstudio.com/items?itemName=hbenl.vscode-test-explorer